zerver/lib/users.py

```diff
@@ -65,6 +65,19 @@ def check_full_name(full_name_raw: str) -> str:
     # ban them.
     if re.search(r"\|\d+$", full_name_raw):
         raise JsonableError(_("Invalid format!"))
+
+    if require_unique_names(realm):
+        normalized_user_full_name = unicodedata.normalize("NFKC", full_name)
+        realm_users_names = UserProfile.objects.filter(realm=realm).values_list(
+            "full_name", flat=True
+        )
+        normalized_realm_users_names = [
+            unicodedata.normalize("NFKC", full_name).lower() for full_name in realm_users_names
+        ]
+
+        if normalized_user_full_name.lower() in normalized_realm_users_names:
+            raise JsonableError(_("User name already exists!"))
```

Comment:
Oh shoot, does this prevent you from changing your name to a different spelling of your own name? It seems like we need to exclude you from the list of users.

Reply:
Thanks for noticing! I never considered this angle

Reply:
Let's also make the error message this: `Unique names required in this organization."`

Reply:
<img width="308" alt="image" src="https://github.com/zulip/zulip/assets/97049524/3898ee68-6b6f-4548-92c0-a3b15e86c56c">

Reply:
lgtm!